packages/visualizations/src/components/Map/MapRender.svelte

```diff
@@ -1,73 +1,85 @@
 <svelte:options immutable={true} />
 
-<script>
-    /*
-TODO:
-- Restrict movement:
-    - Block minZoom to the initial one after fitting the bbox
-    - Compute maxZoom based on the smallest extent when using queryFeature restricted to this bbox
-- Display the map based on style, source, layer, refresh when one changes
-- Adapt display based on the size of the map (single, main, side)
-*/
-    import maplibregl from 'maplibre-gl';
+<script lang="ts">
+    import maplibregl, {
+        Map,
+        SourceSpecification,
+        StyleSpecification,
+        NavigationControl,
+        LngLatBoundsLike,
+        MapSourceDataEvent,
+        MapMouseEvent,
+        LngLatLike,
+    } from 'maplibre-gl';
     import { onMount } from 'svelte';
     import { debounce } from 'lodash';
+    // eslint-disable-next-line import/no-unresolved
+    import type { BBox } from 'geojson';
     import { computeMaxZoomFromGeoJsonFeatures, getFixedTooltips } from './utils';
     import ColorsLegend from '../utils/ColorsLegend.svelte';
+    import type { ColorsScale, DataBounds, LegendVariant } from '../types';
+    import type {
+        ChoroplethFixedTooltipDescription,
+        MapLayer,
+        MapRenderTooltipFunction,
+        MapLegend,
+    } from './types';
 
     // maplibre style (basemap)
-    export let style;
+    export let style: StyleSpecification;
     // maplibre source config
-    export let source;
+    export let source: SourceSpecification;
     // maplibre layer config
-    export let layer;
+    export let layer: MapLayer;
     // bounding box to start from, and restrict to it
-    export let bbox;
-
+    export let bbox: BBox;
     // option to disable map interactions
-    export let interactive;
-
+    export let interactive: boolean;
     // options to display legend
-    export let legend;
-    export let colorsScale;
-    export let dataBounds;
-    let clientWidth;
-    $: legendVariant = clientWidth <= 375 ? 'fluid' : 'fixed';
```

Comment:
Indeed I can do somerhing like
```
const LEGENDVARIANT_FLUID = 'fluid';
const LEGENDVARIANT_FIXED = 'fixed';

export type LegendVariant = typeof LEGENDVARIANT_FLUID | typeof LEGENDVARIANT_FIXED;
```

But I'm not sure of the best place where to put the constants. I'd like to put them into ColorLegend.svelte, but I'm not sure how imports will work, and also this would be a circular import quickly.

What's the trick you're talking about exactly? Maybe it's precisely the answer to my problems, but I don't think I know about it